### ISSUE TITLE
Feature request probability version of `categorical_crossentropy`

### ISSUE THREAD

**Issue by SumNeuron** [keras-team/keras#10993]
As stated in the docs regarding [categorical_crossentropy](https://keras.io/losses/#categorical_crossentropy):

>Note: when using the categorical_crossentropy loss, your targets should be in categorical format (e.g. if you have 10 classes, the target for each sample should be a 10-dimensional vector that is all-zeros except for a 1 at the index corresponding to the class of the sample). In order to convert integer targets into categorical targets, you can use the Keras utility to_categorical:

This is great most of the time. However, what if I want to use soft-labeling?
e.g. if I have three classes then given my input _x_i_ the __target__ output might be `[0.9, 0.05, 0.05]`

**Comment by joelthchao** [comment]
What would happen if you directly provide target in soft-label form?

**Comment by SumNeuron** [comment]
@joelthchao could you elaborate on what you mean?
